105 같이전시 모집글 수정/삭제 구현

leejoonsang opened the pull request:
## :information_desk_person: 간단 소개
함께전시 모집 게시글 수정&삭제 하는 기능

## :heavy_check_mark: 작업 내용 설명
GatheringController, GatheringService -> 로직 추가
GatheringEntity -> 메소드 추가
GatheringControllerTest -> 각각에 대한 성공&실패 테스트코드 추가

## :bulb: 참고사항
확인 부탁드립니다!

leejoonsang committed:
feat: editGathering 메소드 추가

leejoonsang committed:
feat: 수정/삭제 기능 추가

leejoonsang committed:
feat: 수정/삭제 기능 테스트 코드 작성
1. 성공
2. 실패 - 작성자 불일치 / 인증 실패

chlalswns200 commented on src/main/java/com/dev/museummate/service/GatheringService.java:
```diff
@@ -44,4 +60,65 @@ public GatheringDto posts(GatheringPostRequest gatheringPostRequest, String emai
+        );
+
+        return GatheringDto.toDto(savedGathering);
+    }
```
save 쿼리를 날리는 부분이 없는데 영속성 비교로 update 쿼리를 나가게하는 방식인가요?

chlalswns200 commented on src/main/java/com/dev/museummate/service/GatheringService.java:
```diff
@@ -44,4 +60,65 @@ public GatheringDto posts(GatheringPostRequest gatheringPostRequest, String emai
+
+        checkUser(user, savedGathering.getUser());
+
+        gatheringRepository.delete(savedGathering);
```
deletedAt에 기록되는지 실제 db에서 체크하셨나요?

leejoonsang committed:
feat: 수정/삭제 기능 로직 수정

leejoonsang committed:
fix: startsAt/endsAt -> startAt/endAt 수정

leejoonsang committed:
fix: startsAt/endsAt -> startAt/endAt 수정

chlalswns200 committed:
Merge branch 'develop' into 105-같이전시-모집글-수정/삭제-구현

chlalswns200 committed:
refacto: starts,ends -> start,end

chlalswns200 committed:
feat: edit() 검증 로직 추가
maxPeople 수정 할때 현재 값하고 비교하면서
게시글 open close 여부를 다시 결정한다

chlalswns200 commented:
conflict 해결하고 브랜치 받아서 테스트 해봤습니다
s 문제 때문에 s 다 수정하고 돌려봤습니다
수정 로직 에서 게시글 open하는 로직 추가 했습니다.
메인
exhibitionControllerTest
GatheringControllerTest
3개 전부 이상 없이 돌아갔습니다
추가로 수정 필요한 사항 보이시면 말씀해주세요!

geuun committed:
Merge branch 'develop' into 105-같이전시-모집글-수정/삭제-구현

geuun committed:
test: Entity 필드 추가로 인한 TestCode 수정

1. GatheringControllerTest의 생성자를 이용했던 코드를 builder패턴으로
   변경했습니다.
2. ReviewController의 생성자를 이용했던 코드를 builder패턴으로
   변경했습니다.

geuun commented:
추후 리펙토링의 편의를 위해 생성자 패턴에서 Builder 패턴으로 변경했습니다.

geuun reviewed (approved):
수고하셨습니다!

리뷰 완료했습니다 :)

geuun commented:
병합 진행하겠습니다 !
